master/buildbot/reporters/utils.py

```diff
@@ -87,6 +87,32 @@ def getDetailsForBuild(master, build, wantProperties=False, wantSteps=False,
     return ret
 
 
+@defer.inlineCallbacks
+def getDetailsForBuildRequest(master, buildrequest):
+    buildset = yield master.data.get(("buildsets", buildrequest['buildsetid']))
+    builder = yield master.data.get(("builders", buildrequest['builderid']))
+
+    build = dict()
+    build['buildrequest'], build['buildset'] = buildrequest, buildset
```

Comment:
I would prefer separate lines

Reply:
I think it makes sense to use the dict literal syntax ( `{key: 'value', ...}`) here.